src/libraries/Common/tests/TestUtilities/System/PlatformDetection.Unix.cs

```diff
@@ -30,6 +30,7 @@ public static partial class PlatformDetection
 
         public static bool IsMonoLinuxArm64 => IsMonoRuntime && IsLinux && IsArm64Process;
         public static bool IsNotMonoLinuxArm64 => !IsMonoLinuxArm64;
+        public static bool IsNotRiscV64Ubuntu => !IsUbuntu || !IsRiscV64Process;
```

Comment:
If this is working around QEMU specific bugs, it should be named accordingly. 

I assume that these workarounds would be applicable to any platform when running on QEMU.

Reply:
Is this just about naming? Wouldn't it make sense to actually try to detect if the test process is running under QEMU?

Reply:
Yes, I agree

Reply:
https://github.com/dotnet/runtime/pull/102509#issuecomment-2125911845 has some discussion on a possible way to detect QEMU:

```
    // Check for process name of PID 1 indicating emulation
    char cmdline[256];
    FILE *cmdline_file = fopen("/proc/1/cmdline", "r");
    if (cmdline_file != NULL)
    {
        fgets(cmdline, sizeof(cmdline), cmdline_file);
        fclose(cmdline_file);

        if (strstr(cmdline, "qemu") != NULL )
        {
            return true;
        }
    }
```

(Of course, we would want to have managed version of this check.)

cc @am11

Reply:
Agreed that we should isolate QEMU workarounds (and possibly report upstream on gitlab). I can prepare a PR by additing detection in minipal and wire it for P/Invoke.

(my RVA22 MuseBook is shipping by mid-July, I'm looking forward to perform .NET testing on the device 😊)

Reply:
https://github.com/dotnet/runtime/pull/104125, takes care of both system- and process-level emulation detection.

Reply:
> If this is working around QEMU specific bugs, it should be named accordingly.
> 
> I assume that these workarounds would be applicable to any platform when running on QEMU.

You are right, issues were detected on RISC-V Qemu but if I understand Qemu code and reports correctly they are not architecture specific.

Reply:
Now detection is based on environment as discussed: https://github.com/dotnet/runtime/pull/104125#discussion_r1663029220 so I'm resolving this thread.